lib/chef/knife/ec_key_import.rb

```diff
@@ -190,7 +191,17 @@ def import_user_data(path)
               d['hashed_password'] = nil
               d['salt'] = nil
             end
-            users_to_update.update(d)
+            begin
+              users_to_update.update(d)
+            rescue => ex
+              ui.warn "Could not restore user #{d['username']}"
+              if ex.class == Sequel::ForeignKeyConstraintViolation
+                message = "This error usually indicates that a user already exists with a different ID and is associated with one or more organizations on the target system.  The username is #{d['username']} and the ID in the backup files is #{d['id']}"
+                ui.warn message
+                ex = ex.exception "#{ex.message} #{message}"
+              end
+              knife_ec_error_handler.add(ex)
```

Comment:
This code is also called from `knife ec restore` here: https://github.com/chef/knife-ec-backup/blob/master/lib/chef/knife/ec_restore.rb#L187

Perhaps we should have a config option we can set in the calling code to pass in the error handler so we don't end up with two separate error reports (I may be missing why that won't happen in the current code)

Reply:
I'm wondering if it would be better to hard fail in this case since it happens in the early part of the restore and probably means something is wrong.